### DISCUSSION TITLE
[archived] "Enhanced `depends` Logic" docs dicussion

### DISCUSSION THREAD

**Discussion by giscus** [argoproj/argo-workflows#8693] [General]
# argo-workflows/enhanced-depends-logic/

https://argoproj.github.io/argo-workflows/enhanced-depends-logic/

**Comment by czchen** [comment]
Is it possible to have `task.Completed` to indicate the task has been completed, no matter the result?

In our use case, we create a PVC and want to remove it once Workflow is completed. Use `ownerReferences` can only ensure PVC will be removed once Workflows is removed, so we create a cleaner task with `resource.action: delete` to do so. However, in cleaner task `depends`, we need to list all possible results from tasks who need this PVC. At the end, the `depends`  becomes `(x || x.Failed || x.Errored) && (y || y.Failed || y.Errored) && ...`, which causes maintainability problem. It would be much better if there is a catch all result (e.g. `task.Completed`).

**Reply by agilgur5** [reply]
Please file feature requests as issues using the feature request template

**Comment by elliotgunton** [comment]
>For convenience, if an omitted task result is equivalent to `(task.Succeeded || task.Skipped || task.Daemoned)`.

Could this be corrected/rephrased for clarity to
>For convenience, if a task result is not specified it is equivalent to `(task.Succeeded || task.Skipped || task.Daemoned)`.

This avoids using the word "omitted" which is a task result itself

**Reply by agilgur5** [reply]
Agreed. I fixed this in #13110.

Could've submitted a PR yourself though 😉

**Reply by elliotgunton** [reply]
@agilgur5 sorry, never got round to it and forgot! 😅
